File: ios/NHSOnline/NHSOnline/Data/SubService.swift
struct SubService: KnownService, Codable {
    var path: String?
    var queryString: String?
    var javaScriptInteractionMode: JavaScriptInteractionMode
    var menuTab: MenuTab
    var integrationLevel: IntegrationLevel
    var validateSession: Bool

    init(path: String?,
         queryString: String?,
         javaScriptInteractionMode: JavaScriptInteractionMode,
         menuTab: MenuTab,
         integrationLevel: IntegrationLevel,
         validateSession: Bool ) {
        self.path = path
        self.queryString = queryString
        self.javaScriptInteractionMode = javaScriptInteractionMode
        self.menuTab = menuTab
        self.integrationLevel = integrationLevel
        self.validateSession = validateSession
    }
}

File: android/app/src/main/java/com/nhs/online/nhsonline/services/knownservices/SubService.kt
package com.nhs.online.nhsonline.services.knownservices

import com.nhs.online.nhsonline.services.knownservices.enums.JavaScriptInteractionMode
import com.nhs.online.nhsonline.services.knownservices.enums.MenuTab
import com.nhs.online.nhsonline.services.knownservices.enums.ViewMode
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class SubService(
        override var requiresAssertedLoginIdentity: Boolean,
        override var validateSession: Boolean,
        override var menuTab: MenuTab,
        override var viewMode: ViewMode,
        override var javaScriptInteractionMode: JavaScriptInteractionMode,
        override var showSpinner: Boolean,
        var path: String? = null,
        var queryString: String? = null
): KnownService